internal/linter/check_reserved.go

```diff
@@ -0,0 +1,98 @@
+package linter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/skeema/skeema/internal/tengo"
+)
+
+func init() {
+	RegisterRule(Rule{
+		CheckerFunc:     TableBinaryChecker(reservedChecker),
+		Name:            "reserved",
```

Comment:
Since this automatically populates the corresponding option name, I'd lean towards making the Name field slightly more descriptive -- maybe either "reserved-word" (resulting in the option `lint-reserved-word`, which is pretty clear) or perhaps "name-reserved" (which is weirder, but sorts nicely next to [lint-name-case](https://www.skeema.io/docs/options/#lint-name-case))